Local/Factoids.py

```diff
@@ -35,48 +60,11 @@ def handle(self, message):
             target = message['metadata']['source_channel']
             opts = {'target':target}
         except IndexError:
-            logger.error('Could not identify message source in message: %s' % str(message))
+            logger.error('Could not identify message source in message: %s', message)
         command = message['text'].split()[0]
-        if command == '!shrug':
-            txt = '¯\_(ツ)_/¯'
-        elif command == '!tableflip':
-            txt = '(╯°O°）╯︵ ┻━┻'
-        elif command == '!nope':
-            txt = 'https://p.bastelfreak.de/Pfkh/'
-        elif command == '!doit':
-            txt = 'Dooooooooo eeeeeeeettttttttt'
-        elif command == '!wat':
-            txt = 'https://p.bastelfreak.de/1gxL/'
-        elif command == '!@':
-            txt = 'IRC is not Twitter – please do not use @name to address people. use name,' \
-            ' or name: instead. Using @ also prevents you from tab-completing nicks in your IRC client'
-        elif command == '!source':
-            txt = "You can find me on: 'https://github.com/voxpupuli/thevoxfox'"
-        elif command == '!deal':
-            txt = 'https://p.bastelfreak.de/dQ4S/'
-        elif command == '!awesome':
-            txt = 'https://p.bastelfreak.de/tgnla/'
-        elif command == '!nuke':
-            txt = 'https://p.bastelfreak.de/1MQ/'
-        elif command == '!stats':
-            txt = 'http://voxpupuli.bastelfreak.de/'
-        elif command == '!docs':
-            txt = 'https://voxpupuli.org/docs/'
-        elif command == '!http':
-            txt = 'Please supply the full URL! Some of us are lazy'
-        elif command == '!no':
-            txt = 'https://blag.esotericsystems.at/igor/says/no'
-        elif command == '!dog':
-            txt = 'https://p.bastelfreak.de/xtI/'
-        elif command in ['!cat', '!aww', '!awww', '!please']:
-            txt = 'https://p.bastelfreak.de/vi6f05/'
-        elif command == '!dance':
-            txt = "^('-')^ ^('-')^ v('-')v v('-')v <('-'<) (>'-')> <('-'<) (>'-')>"
-        elif command == '!slowclap':
-            txt = 'https://p.bastelfreak.de/yVB9/'
-        elif command == '!eyeballs':
-            txt = 'http://i2.kym-cdn.com/photos/images/original/000/282/317/50d.png'
-        self.reply(message, txt, opts)
+        txt = FACTOIDS.get(command)
```

Comment:
If you're going to use the `get()` method to retrieve the command from the factoids dict, I recommend adding a default value as a failsafe, as that's really the main reason to use the method.

Reply:
Why? ```.get()``` returns None if it's not found and below is a check if it's non-empty. Though in this case it might be better to just catch the KeyError because you're pretty much guaranteed to have a 100% hitrate since another function checks if it exists.

Reply:
Updated to catch the ```KeyError```

Reply:
Returning none makes legobot freakout in this version :(

I fixed in later versions, though.

Reply:
This function has no return statement so I don't see an issue with that.

Reply:
Return was the wrong word, but if you send `None` back to the baseplate in a reply, the baseplate rebroadcasts it and you get blood and guts everywhere. I fixed that in 1.4+ by changing the default behavior for `reply()`. 

Either way, in older versions it's much, much safer to use a safety value for `txt` that you send back with a `reply`.

Reply:
It was this lovely bug: https://github.com/Legobot/Legobot/issues/113

Reply:
It was fixed in this PR: https://github.com/Legobot/Legobot/pull/163

Reply:
Bah, I keep trying to find the PR where I fixed it, but anyways, some history:

reply() used to send the message back to the baseplate, which would then re-advertise the message to all legos. This caused some serious broadcast storms, but moreover, if you sent something back to the baseplate with the `message['text']` as `None`, all the legos would pick it up and try to split `None` (since that's still the defacto pattern for getting the command word out of a message). The result was that all the legos would explode and you'd be left with 100s of lines of debug disaster.

In 1.4, I changed the behavior of reply() to send the message directly back to the lego it originated from (almost always a connector/backend). Now, if you `reply()` to a message, it bypasses the baseplate and does a unicast to the source lego.

Reply:
I had ```if txt: reply()``` so I'd expect that to be safe either way.